### ISSUE TITLE
Turning off built-in Typescript in VSCode loses useful features

### ISSUE THREAD

**Issue by wagenet** [typed-ember/glint#130]
For instance, if I type `assert` it no longer suggests importing from `@ember/debug`.

**Comment by dfreeman** [comment]
It looks like we're not handling completion items with associated code actions (like adding an `import`) correctly.

**Comment by bakerac4** [comment]
I can second this. I lost the ability to remove unused imports. The other thing is that if it does recommend adding an import it always seems to recommend the relative path which is not what I have in my settings

**Comment by wagenet** [comment]
FWIW, I think the only conflict I've seen with built-in TS is the use of private scope in templates. For now I'll stick with protected as we had already been doing and leave built-in TS on.

**Comment by runspired** [comment]
I lost most code-hinting and all hover-for-code-doc abilities.

**Comment by dfreeman** [comment]
> I lost most code-hinting 

Can you give examples of what you mean by code hinting?

The main thing we know we drop on the floor today is what the LSP refers to as "code actions," either affiliated with a completion item (like adding the needed import statement when you pick an identifier) or as standalone directions like cleaning up unused imports. If things that don't fall into that category are missing, that may be something we're unaware of or it may be a bug.

> all hover-for-code-doc abilities.

This sounds like a definite bug, independent of the other things outlined here. Can you file a separate issue with an example of what's failing? (Ideally with some code that I could drop into a Glint-enabled project and see the issue live)

**Comment by dfreeman** [comment]
If anyone is interested adding support for translating the TypeScript language service's code actions (auto-importing new identifiers, removing unused imports, etc) to the LSP equivalent, I'm happy to provide what guidance I can.

**Comment by dfreeman** [comment]
Note that @chadhietala did a massive chunk of work to support code general actions in #499, which is included in recent 1.x beta releases.

We're still missing support for auto-import as part of accepting a completion item (rather than as an after-the-fact independent action), however.

**Comment by NullVoxPopuli** [comment]
Anyone want to implement this?

**Comment by machty** [comment]
Quick 👍  / 👎 : if you're a VS Code user running into this, are you 👍 leaving TS enabled alongside Glint, or 👎 doing without the TS niceties (like auto-import)?

**Comment by wagenet** [comment]
@machty There's a third option here, which is that you turn it on and off as you get annoyed with one or the other 😬

**Comment by wagenet** [comment]
One of the big issues with leaving TS on is that it doesn't recognize .gts files.

**Comment by spuxx1701** [comment]
I have been running both language servers in parallel. It's been the lesser evil for me.

**Comment by NullVoxPopuli** [comment]
With 1.3, you should be able to disable the TS one again

**Comment by NullVoxPopuli** [comment]
I think we can close this!
